components/tag/TagsInput.js

```diff
@@ -0,0 +1,79 @@
+import { useRef } from "react";
+import XMarkIcon from "@heroicons/react/20/solid/XMarkIcon";
+import Input from "../form/Input";
+
+export default function TagsInput({ tags, onTagAdd, onTagRemove }) {
+  const inputRef = useRef(null);
+
+  const keyCode = {
+    comma: 188,
+    backspace: 8,
+  };
+
+  const handleKeyUp = (e) => {
+    const inputValue = inputRef.current.value;
+    const { comma } = keyCode;
+    if (e.keyCode !== comma) return;
+    const newTag = inputValue.trim().replace(/,/g, "");
+    if (!newTag) return;
+    onTagAdd(newTag);
+    inputRef.current.value = "";
+  };
+
+  const handleKeyDown = (e) => {
+    const inputValue = inputRef.current.value;
+    const { backspace } = keyCode;
+    if (e.keyCode === backspace && !inputValue) {
+      const removedTag = tags.pop();
+      onTagRemove(removedTag);
+      inputRef.current.value = removedTag;
+    }
+  };
+
+  const tagItems = tags.map((tag, i) => (
+    <li
+      key={i}
+      className="flex items-center gap-x-1 text-sm p-1 font-mono border rounded-md line-clamp-1"
+    >
+      <span>{tag}</span>
+      <button
+        type="button"
+        className="inline-block text-center"
+        onClick={() => onTagRemove(tag)}
+      >
+        <span className="absolute h-px w-px whitespace-nowrap overflow-hidden">
+          remove {tag} tag
+        </span>
+        <span aria-hidden="true">
+          <XMarkIcon
+            className="w-4 h-4 hover:text-tertiary-medium"
+            title={`Remove ${tag}`}
+          />
+        </span>
```

Comment:
This icon would be read out in screen readers ("cross mark"). Using `aria-hidden` puts a stop to that